src/pythonfinder/models/common.py

```diff
@@ -0,0 +1,22 @@
+from pydantic import BaseModel, Extra
+
+
+class FinderBaseModel(BaseModel):
+    def __setattr__(self, name, value):  # noqa: C901 (ignore complexity)
```

Comment:
Hey @matteius,

could you explain to me why a custom implementation is needed for `__setattr__` and how this implementation is different from the default behavior?

Thanks a lot!

fin swimmer

Reply:
Thanks for your quick response :+1: 

Not sure when you first implement this pattern. But I guess this is no longer needed, when setting `underscore_attrs_are_private` to `True` (https://docs.pydantic.dev/latest/usage/models/#private-model-attributes), right?

Reply:
You're welcome to see if the tests pass in pipenv and this library by refactoring this work further -- I just wanted to get my feet wet in this library and have the larger goal of converting pipenv to a more modern framework.